pkg/pillar/agentlog/agentlog.go

```diff
@@ -119,7 +127,7 @@ func handleSignals(sigs chan os.Signal) {
 			switch sig {
 			case syscall.SIGUSR1:
 				stacks := getStacks(true)
-				log.Warnf("SIGUSR1 triggered stack traces:\n%v\n", stacks)
+				logStacks(fmt.Sprintf("SIGUSR1 triggered stack traces:\n\n %v\n", stacks))
```

Comment:
I would instead print the SIGUSR1 line with a len(stackArray) and then end with some "SIGUSR1 end of stacks"
message. Thus you don't need to concat the line(s) which you include with "SIGUSR1" with \n\n just to split them apart.